### ISSUE TITLE
Inaccurate error "Could not create another Distribution certificate"

### ISSUE THREAD

**Issue by aaronraimist** [fastlane/fastlane#19349]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I was trying to setup fastlane match for the first time. I kept having to rerun it because of various errors. Eventually I got to the error "Could not create another Distribution certificate, reached the maximum number of available Distribution certificates." so I went to https://developer.apple.com/account/resources/certificates/list and revoked a distribution certificate and ran fastlane match again. Same error so I deleted another one. Finally I have deleted them all but continue to receive the same error from fastlane so I ran with `--verbose` and it turns out that was not the error at all.

The actual error was

```
    "status" : "409",
    "code" : "ENTITY_ERROR",
    "title" : "There is a problem with the request entity",
    "detail" : "You already have a current Developer ID Application certificate or a pending certificate request."
```

I still don't quite know what to do with that but at least it is a little more accurate. From https://developer.apple.com/forums/thread/659545 I gather that I can't revoke the developer ID certificate easily and I wish I would have known that before fastlane created more of them.

##### Command executed
`fastlane match developer_id username:"REDACTED" app_identifier:"REDACTED" --verbose`

### Environment

<details>
  <pre> swift-driver version: 1.26.9 <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 11.5.2                                           |
| Ruby                        | 2.7.4                                            |
| Bundler?                    | true                                             |
| Git                         | git version 2.30.1 (Apple Git-130)               |
| Installation Source         | ~/.gem/bin/fastlane                              |
| Host                        | macOS 11.5.2 (20G95)                             |
| Ruby Lib Dir                | /opt/homebrew/Cellar/ruby@2.7/2.7.4/lib          |
| OpenSSL Version             | OpenSSL 1.1.1k  25 Mar 2021                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode-beta.app/Contents/Developer/ |
| Xcode Version               | 13.0                                             |
| Swift Version               | 5.5                                              |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status  |
| -------- | ------- | -------------- |
| fastlane | 2.193.1 | 💥 Check failed |

### Loaded fastlane plugins:

**No plugins Loaded** </pre>
</details>

Related: https://github.com/fastlane/fastlane/issues/19150 https://github.com/fastlane/fastlane/issues/18605

**Comment by aaronraimist** [comment]
This is still a problem.

**Comment by aaronraimist** [comment]
This is still a problem.

**Comment by aaronraimist** [comment]
This is still a problem.

**Comment by marcellodesales** [comment]
👎 still a problem

**Comment by danielRi** [comment]
Same problem here

**Comment by piscespieces** [comment]
problem

**Comment by ashwin-dhakaita** [comment]
Same problem.

**Comment by solikhver** [comment]
I have solved exact the same issue by running `fastlane match nuke` and repeating attempt.

**Comment by colejd** [comment]
Me too thanks

**Comment by aaronraimist** [comment]
@fastlane-bot this is still a problem

**Comment by piscespieces** [comment]
Not sure how I solved this, or if I did something that I can't remember or just got lucky but as far as I can remember right now:

1. Make sure you haven't reach the Distribution Certificates limit (3) in your Apple Developer platform. In case you do, then you gotta make some space for a new one.
2. Delete certificates you won't longer need from your Keychain Access.
3. Try again by running `fastlane match appstore`, that should go ahead and attempt to create a new Distribution Certificate

Let me know if this was helpful at all.
